Drain and  waiting for subscription close  is causing panic: send on closed channel

### Observed behavior

This is our graceful shutdown code, I'm trying to drain and checking if the drain is successful.
```go
unsubCh := sub.StatusChanged(nats.SubscriptionClosed)
		err := sub.Drain()
		if err != nil {
			s.logger.Error("error draining subscription", "error", err, "subject", subject, "queueGroup", queueGroup)
		}
		<-unsubCh
		close(msgCh)
```
Getting the following error:
```
panic: send on closed channel

goroutine 514 [running]:
github.com/nats-io/nats%2ego.(*Subscription).sendStatusEvent(...)
	/home/awcator/go/pkg/mod/github.com/nats-io/nats.go@v1.43.0/nats.go:4665
github.com/nats-io/nats%2ego.(*Subscription).changeSubStatus(...)
	/home/awcator/go/pkg/mod/github.com/nats-io/nats.go@v1.43.0/nats.go:4689
github.com/nats-io/nats%2ego.(*Conn).removeSub(0xc001422708, 0xc0014da0f0)
	/home/awcator/go/pkg/mod/github.com/nats-io/nats.go@v1.43.0/nats.go:4537 +0x286
github.com/nats-io/nats%2ego.(*Conn).checkDrained(0xc001422708, 0xc0014da0f0)
	/home/awcator/go/pkg/mod/github.com/nats-io/nats.go@v1.43.0/nats.go:4767 +0x1df
created by github.com/nats-io/nats%2ego.(*Conn).unsubscribe in goroutine 302
	/home/awcator/go/pkg/mod/github.com/nats-io/nats.go@v1.43.0/nats.go:4849 +0x5f3

Process finished with the exit code 2
```

### Expected behavior

Sometimes it closes properly, sometimes it panics. It looks like some internal channel is closed, and it is writing into it. 

### Server and client version

We are using embedded NATS Go server: 
github.com/nats-io/nats-server/v2 v2.10.29
github.com/nats-io/nats.go v1.43.0

### Host environment

OS: Archlinux


### Steps to reproduce

_No response_

--- a/nats.go
+++ b/nats.go
@@ -33,6 +33,7 @@ import (
 	"path/filepath"
 	"regexp"
 	"runtime"
+	"slices"
 	"strconv"
 	"strings"
 	"sync"
@@ -4686,7 +4687,7 @@ func (s *Subscription) registerStatusChangeListener(status SubStatus, ch chan Su
 // will not block. Lock should be held entering.
 func (s *Subscription) sendStatusEvent(status SubStatus) {
 	for ch, statuses := range s.statListeners {
-		if !containsStatus(statuses, status) {
+		if !slices.Contains(statuses, status) {
 			continue
 		}
 		// only send event if someone's listening
@@ -4694,19 +4695,16 @@ func (s *Subscription) sendStatusEvent(status SubStatus) {
 		case ch <- status:
 		default:
 		}
-		if status == SubscriptionClosed {
+	}
+	// After sending SubscriptionClosed status to all listeners,
+	// close all channels and clear the map to prevent future
+	// sends to closed channels that could cause panics
+	if status == SubscriptionClosed {
+		for ch := range s.statListeners {
 			close(ch)
 		}
+		s.statListeners = nil
 	}
-}
-
-func containsStatus(statuses []SubStatus, status SubStatus) bool {
-	for _, s := range statuses {
-		if s == status {
-			return true
-		}
-	}
-	return false
 }
 
 // changeSubStatus changes subscription status and sends events

